### PR TITLE
Merge dev branch

### PR DESCRIPTION
**Pull request by andriycraft**

### CODE REVIEW

File: src/chat/Colors.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    blue: "§9",
+    green: "§a",
+    red: "§c",
+    light_purple: "§d",
+    yellow: "§e",
+    white: "§f",
+    obfuscated: "§k",
+    cursed: this.obfuscated,
+    bold: "§l",
+    // there is no strikethrough because bedrock does not support it
```
Comment (andriy332) [inline src/chat/Colors.js:24]: remove

File: src/chat/Colors.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    // there is no underline because bedrock does not support it
```
Comment (andriy332) [inline src/chat/Colors.js:25]: remove

File: src/console/Logger.js [hunk 3]
```diff
@@ -24,6 +24,30 @@ class Logger {
                     console.log(`${lang.debug.toUpperCase()} | ${message}`)
                 }
                 break
+            default:
+                break
```
Comment (andriy332) [inline src/console/Logger.js:28]: mabye throw an error or something?

File: src/console/Logger.js [hunk 4]
```diff
@@ -24,6 +24,30 @@ class Logger {
...
+            case 'err':
+                console.log(`${Colors.CONSOLE_RED}${lang.error.toUpperCase()}${Colors.CONSOLE_RESET} | ${prefix}${plugin}${suffix} ${message}`)
+                break
+            case 'debug':
+                if (config.debug) {
+                    console.log(`${lang.debug.toUpperCase()} | ${message}`)
+                }
+                break
             default:
                 throw new Error(`${lang.invalid}`)
```
Comment (andriy332) [inline src/console/Logger.js:52]: console.log(`${lang.invalid}`)

File: src/player/KickPlayer.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class KickPlayer {
+
+    constructor() { }
+
+    kick(client, reason = 'You were disconnected') {
+        if (!client) throw new Error("kick(): No client!")
```
Comment (andriy332) [inline src/player/KickPlayer.js:6]: kick(client): No client!

File: src/plugins/Unloader.js [hunk 6]
```diff
@@ -1,28 +1,33 @@
...
             });
-            Logger.prototype.log('Done shutting down plugins!')
+            Logger.prototype.log(lang.doneshuttingdownplugins)
         });
         setTimeout(() => {
-            Logger.prototype.log('Done')
-            process.exit(0)
-        }, 10000)
+            Logger.prototype.log(lang.doneshuttingdown)
+            process.exit(config.exitstatuscode) // TODO: Not sure if this is useful (i mean the config for exit status code)
```
Comment (andriy332) [inline src/plugins/Unloader.js:29]: it is

**Review by andriy332 — CHANGES_REQUESTED** [review]

File: src/plugins/Unloader.js (hunk 6 above)
Comment (andriycraft) [reply src/plugins/Unloader.js:29]: not sure if it is. any examples of usage?

File: src/plugins/Unloader.js [hunk 7]
```diff
@@ -1,28 +1,33 @@
...
-            Logger.prototype.log('The server has 10 sec to shutdown the plugins. If the servers fails to do it in time - there may be some issues with plugin data', 'warning')
+            Logger.prototype.log(lang.shuttingdownplugins)
+            Logger.prototype.log(lang.tensecwarn, 'warning')
             plugins.forEach(plugin => {
                     if (require(`../../plugins/${plugin}`).prototype.getName() === "") {
-                        throw new Error(`Failed to shutdown plugin ${plugin}. No plugin name!`)
+                        throw new Error(lang.failedtoshutdownplugin.replace('%plugin%', plugin))
                     }
 
+                    // TODO: Somehow use await
```
Comment (andriycraft) [inline src/plugins/Unloader.js:22]: TODO: Use await

File: src/player/KickPlayer.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+class KickPlayer {
+
+    constructor() { }
+
+    kick(client, reason = 'You were disconnected') {
```
Comment (andriycraft) [inline src/player/KickPlayer.js:5]: TODO: Make the default kick reason customizable

File: src/player/KickPlayer.js (hunk 5 above)
Comment (andriycraft) [inline src/player/KickPlayer.js]: TODO: Same here

File: src/player/Chatmessage.js [hunk 9]
```diff
@@ -16,6 +19,16 @@ class Chatmessage {
...
     }
 
+    broadcastMessage(msg) {
+        if (!msg) throw new Error("broadcastMessage(msg): No chat message!");
+
+        for (let i = 0; i < PlayerInfo.prototype.getPlayers().length; i++) {
+            let client = PlayerInfo.prototype.getPlayers()[i]
+            this.sendMessage(client, msg)
+        }
+        Logger.prototype.log(`(broadcast) ${msg}`)
```
Comment (andriycraft) [inline src/player/Chatmessage.js:29]: TODO: Translate to other langs

File: src/player/Chatmessage.js [hunk 10]
```diff
@@ -16,6 +19,16 @@ class Chatmessage {
         })
     }
 
+    broadcastMessage(msg) {
+        if (!msg) throw new Error("broadcastMessage(msg): No chat message!");
```
Comment (andriycraft) [inline src/player/Chatmessage.js:23]: TODO: Translate to other langs

File: src/player/Chatmessage.js [hunk 11]
```diff
@@ -1,10 +1,13 @@
...
+
 class Chatmessage {
 
     constructor() { }
 
     sendMessage(client, msg) {
-        if (!msg) throw new Error("sendmessage(): No chat message")
-        if (!client) throw new Error("sendmessage(): No client!")
+        if (!msg) throw new Error("sendmessage(client, msg): No chat message")
+        if (!client) throw new Error("sendmessage(client, msg): No client!")
```
Comment (andriycraft) [inline src/player/Chatmessage.js:10]: TODO: Translate to other langs

File: src/player/Chatmessage.js [hunk 12]
```diff
@@ -1,10 +1,13 @@
+const PlayerInfo = require('../player/PlayerInfo')
+const Logger = require('../console/Logger')
+
 class Chatmessage {
 
     constructor() { }
 
     sendMessage(client, msg) {
```
Comment (andriycraft) [inline src/player/Chatmessage.js:8]: TODO: Translate to other langs

**Comment by andriycraft** [comment]
Update finished

**Pull request merged**